statefun-flink/statefun-flink-datastream/src/main/java/org/apache/flink/statefun/sdk/TypedValue.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.statefun.sdk;
+
+import java.util.Arrays;
+import java.util.Objects;
+import org.apache.flink.api.common.typeinfo.TypeInfo;
+import org.apache.flink.statefun.sdk.types.TypedValueTypeInfoFactory;
+
+/** A typed value containing a {@link TypeName} and serialized bytes. */
+@TypeInfo(TypedValueTypeInfoFactory.class)
+@SuppressWarnings("unused")
+public class TypedValue {
```

Comment:
The purpose of this class is to avoid exposing the generated protobuf TypedValue in the public API. Let me know if you believe I'm over thinking this.